packages/twenty-front/src/modules/object-record/record-merge/hooks/useMergeRecordsContainerTabs.ts

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+
+import { useLingui } from '@lingui/react/macro';
+
+import { IconArrowMerge, IconSettings } from 'twenty-ui/display';
+
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { SingleTabProps } from '@/ui/layout/tab-list/types/SingleTabProps';
+import { MergeRecordsTabId } from '../types/MergeRecordsTabId';
+import {
+  getPositionNumberIcon,
+  getPositionWordLabel,
+} from '../utils/recordMergeUtils';
+
+export const useMergeRecordsContainerTabs = (
+  selectedRecords: ObjectRecord[],
+  loading: boolean,
+): { tabs: SingleTabProps[] } => {
+  const { t } = useLingui();
+
+  const tabs = useMemo(() => {
```

Comment:
Not sure if a use memo is needed here. In the codebase we try to avoid optimization like that, it's duct tape which can hide a real root-cause problem. We want to fix root-cause/re-renders instead